examples/lora/lorawan/basic-demo/main.go

```diff
@@ -0,0 +1,122 @@
+// Simple code for connecting to Lorawan network and uploading sample payload
+
+package main
+
+import (
+	"machine"
+	"time"
+
+	"tinygo.org/x/drivers/examples/lora/lorawan/common"
+	"tinygo.org/x/drivers/lora"
+	"tinygo.org/x/drivers/lora/lorawan"
+)
+
+const (
+	LORAWAN_RECONNECT_DELAY_SEC = 60
+)
+
+// change these to test a different UART or pins if available
+var (
+	uart  = machine.Serial
+	tx    = machine.UART_TX_PIN
+	rx    = machine.UART_RX_PIN
+	input = make([]byte, 0, 64)
+
+	radio   lora.Radio
+	session *lorawan.Session
+	otaa    *lorawan.Otaa
+
+	defaultTimeout uint32 = 1000
+
+	lorawanConnected bool
+	lorawanDoConnect bool
+)
+
+// parse() processes serial console commands
+func parse(data []byte) error {
+	s := string(data)
+
+	if s == "CONNECT" {
+		lorawanDoConnect = true
+	}
+	return nil
+}
+
+// This will be run as a go routine to keep us connected
```

Comment:
Perhaps I am reading this wrong, but looks like this function would retry until it connects once, but then once connected would not reconnect. Perhaps the comment just needs to explain this.